contrib/bindings/go/pathrs/pathrs.go

```diff
@@ -0,0 +1,328 @@
+func (r *Root) Resolve(path string) (*Handle, error) {
+	handler := C.pathrs_resolve(r.root, C.CString(path))
+	if handler == nil {
+		return nil, handleErr(C.PATHRS_ROOT, unsafe.Pointer(r.root))
+	}
+
+	return newHandle(handler), nil
+}
+
+// Creat creates a file with a such mode by path according with the root path
+func (r *Root) Creat(path string, mode uint) (*Handle, error) {
+	handler := C.pathrs_creat(r.root, C.CString(path), C.uint(mode))
+	if handler == nil {
+		return nil, handleErr(C.PATHRS_ROOT, unsafe.Pointer(r.root))
+	}
+
+	return newHandle(handler), nil
+}
+
+// Rename Within the given root's tree, perform the rename of src to dst,
+// or change flags on this file if the names are the same it's only change the flags
+func (r *Root) Rename(src, dst string, flags int) error {
+	C.pathrs_rename(r.root, C.CString(src), C.CString(dst), C.int(flags))
+	return handleErr(C.PATHRS_ROOT, unsafe.Pointer(r.root))
+}
+
+// Mkdir creates a directory with a such mode by path
+func (r *Root) Mkdir(path string, mode uint) error {
+	C.pathrs_mkdir(r.root, C.CString(path), C.uint(mode))
+	return handleErr(C.PATHRS_ROOT, unsafe.Pointer(r.root))
+}
+
+// Mknod creates a filesystem node named path
+// with attributes mode and dev
+func (r *Root) Mknod(path string, mode uint, dev int) error {
+	C.pathrs_mknod(r.root, C.CString(path), C.uint(mode), C.dev_t(dev))
+	return handleErr(C.PATHRS_ROOT, unsafe.Pointer(r.root))
+}
+
+// Hardlink creates a hardlink of file named target and place it to path
+func (r *Root) Hardlink(path, target string) error {
+	C.pathrs_hardlink(r.root, C.CString(path), C.CString(target))
+	return handleErr(C.PATHRS_ROOT, unsafe.Pointer(r.root))
+}
+
+// Symlink creates a symlink of file named target and place it to path
+func (r *Root) Symlink(path, target string) error {
+	C.pathrs_symlink(r.root, C.CString(path), C.CString(target))
+	return handleErr(C.PATHRS_ROOT, unsafe.Pointer(r.root))
+}
+
+// IntoFd unwraps a file-based libpathrs object to obtain its underlying file
+// descriptor.
+//
+// It is critical that you do not operate on this file descriptor yourself,
+// because the security properties of libpathrs depend on users doing all
+// relevant filesystem operations through libpathrs.
+func (r *Root) IntoFd() (int, error) {
+	fd := int(C.pathrs_into_fd(C.PATHRS_ROOT, unsafe.Pointer(r.root)))
+	if fd < 0 {
+		return 0, handleErr(C.PATHRS_ROOT, unsafe.Pointer(r.root))
+	}
+
+	return fd, nil
+}
+
+// Clone creates a copy of root handler the new object will have a separate lifetime
+// from the original, but will refer to the same underlying file
+func (r *Root) Clone() (*Root, error) {
+	newRoot := (*C.pathrs_root_t)(C.pathrs_duplicate(C.PATHRS_ROOT, unsafe.Pointer(r.root)))
+	err := handleErr(C.PATHRS_ROOT, unsafe.Pointer(newRoot))
+	if err != nil {
+		return nil, err
+	}
+
+	return &Root{root: newRoot}, nil
+}
+
+// Free frees underling caught resources
+func (r *Root) Free() {
```

Comment:
This (and `handle.Free`) should be called `Close` to match most other Go libraries.